python/extension/CMakeLists.txt

```diff
@@ -0,0 +1,149 @@
+# ============================================================================ #
```

Comment:
Is `python/extension` a new feature? (Separate PR?)

Reply:
No, I just moved the extension related files into their own subfolder to be able to configure cmake for them without impacting other parts of the folder.